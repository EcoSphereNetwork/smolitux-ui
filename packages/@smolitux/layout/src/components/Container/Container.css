.smolitux-container {
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  box-sizing: border-box;
}

/* Responsive Anpassungen */
@media (max-width: 576px) {
  .smolitux-container {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media (min-width: 577px) and (max-width: 768px) {
  .smolitux-container {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media (min-width: 769px) {
  .smolitux-container {
    padding-left: 2rem;
    padding-right: 2rem;
  }
}