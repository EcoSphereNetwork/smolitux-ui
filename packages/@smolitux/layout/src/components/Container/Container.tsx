// packages/@smolitux/layout/src/components/Container/Container.tsx
import React, { forwardRef } from 'react';

export interface ContainerProps extends React.HTMLAttributes<HTMLDivElement> {
  /** Maximale Breite des Containers */
  maxWidth?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' | 'full' | 'none';
  /** Horizontales Padding deaktivieren */
  disableGutters?: boolean;
  /** Container auf Bildschirmhöhe setzen */
  fullHeight?: boolean;
  /** Content innerhalb des Containers zentrieren */
  centerContent?: boolean;
}

/**
 * Container-Komponente für konsistentes Layout in verschiedenen Breakpoints
 * 
 * @example
 * ```tsx
 * <Container maxWidth="lg">
 *   <h1>Meine Seite</h1>
 *   <p>Inhalt, der auf verschiedenen Bildschirmgrößen konsistent angezeigt wird</p>
 * </Container>
 * ```
 */
export const Container = forwardRef<HTMLDivElement, ContainerProps>(({
  maxWidth = 'lg',
  disableGutters = false,
  fullHeight = false,
  centerContent = false,
  className = '',
  children,
  ...rest
}, ref) => {
  // Maximale Breiten-Klassen entsprechend der maxWidth-Prop
  const maxWidthClasses = {
    xs: 'max-w-sm', // 640px
    sm: 'max-w-md', // 768px
    md: 'max-w-lg', // 1024px
    lg: 'max-w-3xl', // 1280px
    xl: 'max-w-5xl', // 1536px
    '2xl': 'max-w-7xl', // 1920px
    full: 'max-w-full',
    none: ''
  };

  // Basis-Klassen
  const containerClasses = [
    // Basis-Container-Klassen
    'w-full mx-auto',
    
    // Max-Width basierend auf der Prop
    maxWidthClasses[maxWidth],
    
    // Gutters (horizontales Padding)
    disableGutters ? '' : 'px-4 sm:px-6 md:px-8',
    
    // Vollständige Höhe
    fullHeight ? 'h-full' : '',
    
    // Content zentrieren (flexbox)
    centerContent ? 'flex flex-col items-center justify-center' : '',
    
    // Benutzerdefinierte Klassen
    className
  ].filter(Boolean).join(' ');

  return (
    <div
      ref={ref}
      className={containerClasses}
      {...rest}
    >
      {children}
    </div>
  );
});

Container.displayName = 'Container';

export default Container;
