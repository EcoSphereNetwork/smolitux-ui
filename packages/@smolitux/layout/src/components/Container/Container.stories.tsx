import React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { Container } from './Container';

const meta: Meta<typeof Container> = {
  title: 'Layout/Container',
  component: Container,
  parameters: {
    layout: 'fullscreen',
  },
  tags: ['autodocs'],
  argTypes: {
    maxWidth: {
      control: { type: 'select' },
      options: ['xs', 'sm', 'md', 'lg', 'xl', '2xl', 'full', 'none'],
    },
    disableGutters: { control: { type: 'boolean' } },
    fullHeight: { control: { type: 'boolean' } },
    centerContent: { control: { type: 'boolean' } },
  },
};

export default meta;
type Story = StoryObj<typeof Container>;

// Hilfsfunktion zum Erstellen von Beispielinhalten
const ExampleContent = () => (
  <div className="p-4 bg-gray-100 dark:bg-gray-800 rounded-lg">
    <h2 className="text-xl font-bold mb-4">Container-Inhalt</h2>
    <p className="mb-4">
      Dieser Container passt seine maximale Breite basierend auf der <code>maxWidth</code>-Prop an.
      Probieren Sie verschiedene Werte aus, um zu sehen, wie sich der Container verh√§lt.
    </p>
    <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
      <div className="bg-white dark:bg-gray-700 p-4 rounded shadow">
        <h3 className="font-semibold mb-2">Abschnitt 1</h3>
        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
      </div>
      <div className="bg-white dark:bg-gray-700 p-4 rounded shadow">
        <h3 className="font-semibold mb-2">Abschnitt 2</h3>
        <p>Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>
      </div>
    </div>
  </div>
);

// Hilfsfunktion zum Anzeigen der Container-Grenzen
const ContainerWithBorder = (props: React.ComponentProps<typeof Container>) => (
  <div className="bg-gray-200 dark:bg-gray-900 min-h-[300px] p-4">
    <div className="border-2 border-dashed border-blue-500 dark:border-blue-400">
      <Container {...props} />
    </div>
  </div>
);

export const Default: Story = {
  render: (args) => (
    <ContainerWithBorder {...args}>
      <ExampleContent />
    </ContainerWithBorder>
  ),
  args: {
    maxWidth: 'lg',
    disableGutters: false,
    fullHeight: false,
    centerContent: false,
  },
};

export const Small: Story = {
  render: (args) => (
    <ContainerWithBorder {...args}>
      <ExampleContent />
    </ContainerWithBorder>
  ),
  args: {
    maxWidth: 'sm',
    disableGutters: false,
  },
};

export const Medium: Story = {
  render: (args) => (
    <ContainerWithBorder {...args}>
      <ExampleContent />
    </ContainerWithBorder>
  ),
  args: {
    maxWidth: 'md',
    disableGutters: false,
  },
};

export const Large: Story = {
  render: (args) => (
    <ContainerWithBorder {...args}>
      <ExampleContent />
    </ContainerWithBorder>
  ),
  args: {
    maxWidth: 'lg',
    disableGutters: false,
  },
};

export const ExtraLarge: Story = {
  render: (args) => (
    <ContainerWithBorder {...args}>
      <ExampleContent />
    </ContainerWithBorder>
  ),
  args: {
    maxWidth: 'xl',
    disableGutters: false,
  },
};

export const FullWidth: Story = {
  render: (args) => (
    <ContainerWithBorder {...args}>
      <ExampleContent />
    </ContainerWithBorder>
  ),
  args: {
    maxWidth: 'full',
    disableGutters: false,
  },
};

export const WithoutGutters: Story = {
  render: (args) => (
    <ContainerWithBorder {...args}>
      <ExampleContent />
    </ContainerWithBorder>
  ),
  args: {
    maxWidth: 'lg',
    disableGutters: true,
  },
};

export const CenteredContent: Story = {
  render: (args) => (
    <ContainerWithBorder {...args}>
      <div className="bg-white dark:bg-gray-700 p-6 rounded-lg shadow-md max-w-md">
        <h2 className="text-xl font-bold mb-4 text-center">Zentrierter Inhalt</h2>
        <p className="text-center">
          Dieser Container verwendet <code>centerContent</code>, um seinen Inhalt
          horizontal und vertikal zu zentrieren.
        </p>
      </div>
    </ContainerWithBorder>
  ),
  args: {
    maxWidth: 'lg',
    centerContent: true,
    fullHeight: true,
  },
};