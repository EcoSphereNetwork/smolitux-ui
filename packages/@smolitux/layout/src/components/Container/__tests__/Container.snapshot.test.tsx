import React from 'react';
import { render } from '@testing-library/react';
import { Container } from '../Container';

describe('Container Snapshots', () => {
  it('renders default container', () => {
    const { asFragment } = render(<Container>Content</Container>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with options', () => {
    const { asFragment } = render(
      <Container maxWidth="xl" centerContent fullHeight disableGutters>
        <p>Centered</p>
      </Container>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
