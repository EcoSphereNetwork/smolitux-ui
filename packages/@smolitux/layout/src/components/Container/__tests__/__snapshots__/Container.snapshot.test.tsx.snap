// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Snapshots renders default container 1`] = `
<DocumentFragment>
  <div
    class="w-full mx-auto max-w-3xl px-4 sm:px-6 md:px-8"
  >
    Content
  </div>
</DocumentFragment>
`;

exports[`Container Snapshots renders with options 1`] = `
<DocumentFragment>
  <div
    class="w-full mx-auto max-w-5xl h-full flex flex-col items-center justify-center"
  >
    <p>
      Centered
    </p>
  </div>
</DocumentFragment>
`;
